source/jormungandr/jormungandr/interfaces/v1/Journeys.py

```diff
@@ -294,9 +294,18 @@ def wrapper(*args, **kwargs):
                 return objects
             for journey in objects[0]['journeys']:
                 args = dict(request.args)
-                ids = {o['stop_point']['stop_area']['id']
+                ids = {o['stop_point']['id']
                        for s in journey.get('sections', []) if 'from' in s
                        for o in (s['from'], s['to']) if 'stop_point' in o}
+
+                #only one first_section_mode is allowed if present in the request
+                #same for last_section_mode also
+                if 'first_section_mode[]' in args:
+                    args['first_section_mode[]'] = journey['sections'][0]['mode']
+                if 'last_section_mode[]' in args:
+                    args['last_section_mode[]'] = journey['sections'][-1]['mode']
+                args['min_nb_transfers'] = journey['nb_transfers']
```

Comment:
It will not be enough for direct  journey, as there is 0 transfers in direct pt journeys and non pt journeys.